docs/explainer.md

```diff
@@ -1,137 +1,1058 @@
+
+```js
+// home.json
+{
+  "usingComponents": {
+     "custom": "/components/custom/custom"
+   }
+}
+```
+
+And declaration in the SWAN document using the ID in the configuration file.
+
+```xml
+<!-- home.swan -->
+<view>
+   <custom name="swanapp"></custom>
+</view>
+```
+
+[More information](https://smartprogram.baidu.com/docs/develop/framework/dev/).
+
+
+#### Component properties (definition)
+
+How the properties (attributes) of the components are defined in each MiniApp version. 
+
+##### Quick App (Xiaomi, Huawei)
+
+Components define properties in a `props` array within the object declared in the `<script>` section.     
+
+```html
+<!-– comp.ux -->
+<script>
+  export default {
+    props: ['say', 'propObject'],
+    onInit() {
+      console.info(this.say);    
+      console.info(this.propObject)
+    }
+  }
+</script>
+```
+[More information](https://doc.quickapp.cn/tutorial/framework/parent-child-component-communication.html).
+
+##### Alipay Mini Program
+
+Definition of the component using `Component()` and the properties as pairs in the `props` object.
+
+```js
+// /components/index/index.js
+Component ({
+  props : { 
+    name : 'My name'}
+});
+```
+[More information](https://opendocs.alipay.com/mini/framework/component_object).
+
+
+##### Baidu Smart Mini Program
+
+Definition of the component using `Component()` and the properties in the `properties` object.
+
+```js
+// (custom.js)
+Component({
+  properties: {
+    name: {
+      type: String,
+      value: 'swan',
+    }
+  }
+})
+```
+
+[More information](https://smartprogram.baidu.com/docs/develop/framework/custom-component/).
+
+
+#### Component arguments
+
+How the properties (attributes) are passed to the components when new instances are created. 
+
+##### Quick App (Xiaomi, Huawei)
+
+Use the defined `props` directly as instance attributes.
+
+```xml
+<import name="comp" src="./comp"></import>
+
+<template>
+  <div>
+    <comp 
+      say="First Test" 
+      prop-object="{{obj}}">
+    </comp>
+  </div>
+</template>
+```
+
+[More information](https://doc.quickapp.cn/tutorial/framework/parent-child-component-communication.html).	
+
+
+##### Alipay Mini Program
+
+Pass the properties as instance attributes.
+
+```xml
+<!-- /pages/index/index.axml -->
+<comp name='Another name'></comp> 
+```
+
+[More information](https://opendocs.alipay.com/mini/framework/component_object).	
+
+##### Baidu Smart Mini Program
+
+Properties as instance attributes
+
+```js
+// home.json
+{
+  "usingComponents": {
+     "custom": "/components/custom/custom"
+   }
+}
+```
+
+```xml
+// .swan document
+<view>
+    <custom name="Passing text" >
+    </custom>
+</view>
+```
+
+[More information](https://smartprogram.baidu.com/docs/develop/framework/custom-component/).
+
+
+#### Data binding
+
+All the MiniApp versions have [`{{moustache}}` notation](https://en.wikipedia.org/wiki/Mustache_%28template_system%29) to bind variables from the logic part of the component.
```

Comment:
Might be worth pointing out if the binding is one-way or two-way.

Reply:
Yes, I agree. Thanks. I've included the suggestion at e998f4a22f9d30da4ed1911bd5ac9c792ab3092f